plugins/terraform/resource_load_balancer_test.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/common/models"
+	"testing"
+)
+
+func TestLoadBalancerCreate_basic(t *testing.T) {
+	ctrl, mockClient, provider := setupUnitTest(t)
+	defer ctrl.Finish()
+
+	loadBalancerResource := provider.ResourcesMap["layer0_load_balancer"]
+	d := schema.TestResourceDataRaw(t, loadBalancerResource.Schema, map[string]interface{}{
+		"name":        "test-lb",
+		"environment": "test-env",
+	})
+
+	mockClient.EXPECT().
+		CreateLoadBalancer("test-lb", "test-env", models.HealthCheck{"TCP:80", 30, 5, 2, 2}, []models.Port{}, true).
```

Comment:
Oooh, good call - I can update this with a specified port, but bigger-picture: should I be making CLI-based failure tests? They seem a little useless to me, since the client's mocked out.

 - send off a config that's wrong
 - tell the mockClient to return something saying that it's wrong
 - assert that... I've told the mockClient to say that something's wrong...?